frontend/src/main/java/com/google/tripmeout/frontend/storage/InMemoryTripModelStorage.java

```diff
@@ -0,0 +1,85 @@
+package com.google.tripmeout.frontend.storage;
+
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripAlreadyExistsException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class InMemoryTripModelStorage {
+  Map<String, TripModel> storage = new ConcurrentHashMap<>();
+
+  public void addTrip(TripModel trip) throws TripAlreadyExistsException {
```

Comment:
Did you already talk with @KoopaKing about making our custom exceptions extend `RuntimeException`  versus `Exception`? This is fine as it is, but it's important to have considered this and it largely depends on how it's consumed in the frontends.

In particular, if the servlets will just catch all instances of `TripMeOutException` at the top-level and perform automatic exception translation (which I'd probably recommend), then you may want to consider making this a `RuntimeException`. In that case, you can drop the `throws` clauses on all methods here. That might especially make sense if there there are multiple, intermediate-stage callers who cannot/do not know how to handle the exceptions thrown here.